### PR TITLE
add assertions for new SpaceCounter

### PR DESCRIPTION
**Pull request by mfrancisc**
Paired with: [host-operator-pr](https://github.com/codeready-toolchain/host-operator/pull/701)

work related to jira item: [jira ticket](https://issues.redhat.com/browse/ASC-112)

### CODE REVIEW

File: testsupport/wait/host.go [hunk 1]
```diff
@@ -563,6 +563,46 @@ func (a *HostAwaitility) WaitForTestResourcesCleanup(initialDelay time.Duration)
 				return false, nil
 			}
 		}
+
+		spaceBindingsList := &toolchainv1alpha1.SpaceBindingList{}
```
Comment (xcoulon) [inline testsupport/wait/host.go:567]: should be named `spaceBindingList` but it's a minor comment

**Review by xcoulon — APPROVED** [review]

File: testsupport/wait/host.go [hunk 2]
```diff
@@ -539,7 +539,7 @@ func UntilUserSignupHasStateLabel(expected string) UserSignupWaitCriterion {
 	}
 }
 
-// WaitForTestResourcesCleanup waits for all UserSignup and MasterUserRecord deletions to complete
+// WaitForTestResourcesCleanup waits for all UserSignup, MasterUserRecord, Space(biding), NStemplateSet and Namespace deletions to complete
```
[the remark this answers is not in the document]
Comment (mfrancisc) [reply testsupport/wait/host.go]: fixed in [4c30f44df](https://github.com/mfrancisc/toolchain-e2e/commit/4c30f44df8198939255740d6fb574b80e1db19bf)

File: testsupport/wait/host.go (hunk 1 above)
Comment (mfrancisc) [reply testsupport/wait/host.go]: fixed in [4c30f44df](https://github.com/mfrancisc/toolchain-e2e/commit/4c30f44df8198939255740d6fb574b80e1db19bf)

File: testsupport/wait/host.go [hunk 3]
```diff
@@ -539,7 +539,7 @@ func UntilUserSignupHasStateLabel(expected string) UserSignupWaitCriterion {
 	}
 }
 
-// WaitForTestResourcesCleanup waits for all UserSignup and MasterUserRecord deletions to complete
+// WaitForTestResourcesCleanup waits for all UserSignup, MasterUserRecord, Space(binding), NStemplateSet and Namespace deletions to complete
```
Comment (rajivnathan) [inline testsupport/wait/host.go:542]:
```suggestion
// WaitForTestResourcesCleanup waits for all UserSignup, MasterUserRecord, Space, SpaceBinding, NSTemplateSet and Namespace deletions to complete
```

File: test/metrics/metrics_test.go [hunk 4]
```diff
@@ -56,6 +56,8 @@ func TestMetricsWhenUsersDeactivated(t *testing.T) {
 	metricsAssertion.WaitForMetricDelta(UserSignupsDeactivatedMetric, 0)                                                 // none deactivated
 	metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", memberAwait.ClusterName)                  // none activated on member-1
 	metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 2, "cluster_name", memberAwait2.ClusterName)                 // all activated on member-2
+	metricsAssertion.WaitForMetricDelta(SpacesMetric, 0, "cluster_name", memberAwait.ClusterName)
+	metricsAssertion.WaitForMetricDelta(SpacesMetric, 2, "cluster_name", memberAwait2.ClusterName)
```
Comment (rajivnathan) [inline test/metrics/metrics_test.go:60]:
```suggestion
	metricsAssertion.WaitForMetricDelta(SpacesMetric, 0, "cluster_name", memberAwait.ClusterName)
	metricsAssertion.WaitForMetricDelta(SpacesMetric, 2, "cluster_name", memberAwait2.ClusterName) // 2 spaces created on member-2
```

File: test/metrics/metrics_test.go [hunk 5]
```diff
@@ -79,6 +81,8 @@ func TestMetricsWhenUsersDeactivated(t *testing.T) {
 	metricsAssertion.WaitForMetricDelta(UserSignupsDeactivatedMetric, 2)                                                 // all deactivated
 	metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", memberAwait.ClusterName)                  // all deactivated on member-1
 	metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", memberAwait2.ClusterName)                 // all deactivated on member-2
+	metricsAssertion.WaitForMetricDelta(SpacesMetric, 0, "cluster_name", memberAwait.ClusterName)
+	metricsAssertion.WaitForMetricDelta(SpacesMetric, 0, "cluster_name", memberAwait2.ClusterName)
```
Comment (rajivnathan) [inline test/metrics/metrics_test.go:85]:
```suggestion
	metricsAssertion.WaitForMetricDelta(SpacesMetric, 0, "cluster_name", memberAwait2.ClusterName) // 2 spaces deleted from member-2
```

**Pull request merged**